### PR TITLE
Updated Rules to use "OriginalFileName" Field (First Batch)

### PR DESCRIPTION
**Pull request by nasbench**
- Updated some process_creation rules to use both Image and OriginalFileName as increased accuracy.
- Added another option for the rule "proc_creation_win_powershell_defender_exclusion"
- I'm planning to modify the rest of the rules in small batches if this PR gets approved :)

### CODE REVIEW

**Comment by frack113** [comment]
Hello,
For process_creation I did use only OriginalFileName  because of eventid 4688, so need to keep both.
In proc_creation_win_webshell_detection.yml , you simply exchange `image` by `OriginalFileName`.
I did not check all .

image


`OriginalFileName` have no path , there is no need of `|contains` .

File: rules/windows/process_creation/proc_creation_win_vmtoolsd_susp_child_process.yml [hunk 1]
```diff
@@ -24,13 +24,13 @@ logsource:
...
     selection:
         ParentImage|endswith: '\vmtoolsd.exe'
-        Image|endswith:
-            - '\cmd.exe'
-            - '\powershell.exe'
-            - '\rundll32.exe'
-            - '\regsvr32.exe'
-            - '\wscript.exe'
-            - '\cscript.exe'
+        OriginalFileName|contains:
```
Comment (Neo23x0) [inline rules/windows/process_creation/proc_creation_win_vmtoolsd_susp_child_process.yml:27]: We still need the "Image|endswith" here

File: rules/windows/process_creation/proc_creation_win_uac_wsreset.yml [hunk 2]
```diff
@@ -6,15 +6,15 @@ author: E.M. Anhaus (originally from Atomic Blue Detections, Tony Lambert), oscd
...
+modified: 2022/05/12
 logsource:
   category: process_creation
   product: windows
 detection:
   selection:
     ParentImage|endswith: '\wsreset.exe'
   filter:
-    Image|endswith: '\conhost.exe'
+    OriginalFileName|contains: 'CONHOST.EXE'
```
Comment (Neo23x0) [inline rules/windows/process_creation/proc_creation_win_uac_wsreset.yml:17]: We still need the Images|endswith here

**Comment by Neo23x0** [comment]
Yes, in some cases the Image|endswith has been replaced not extended. We still need the Images|endswith expression. 

And the `OriginalFileName` should be used without `contains`, especially but not only because it could lead to false positives. A 1:1 match is also faster in comparison than a match using `*whoami.exe*` in field values.

**Comment by nasbench** [comment]
I've kept both "OriginalFileName" and "Image" because sometimes built-in tools such as "whoami" are available from other vendors but without "OriginalFileName" field. 

So the idea is to keep these rules detecting binaries with these names and perhaps we could add renamed version in the future to take into account the same args but different names? 

I will fix the errors you pointed out and add the "Image" and remove contains from "OriginalFileName"

**Comment by Neo23x0** [comment]
Looks good to me

File: rules/windows/file_rename/file_rename_win_not_dll_to_dll.yml [hunk 3]
```diff
@@ -7,18 +7,18 @@ references:
...
 detection:
     to_dll:
         TargetFilename|endswith: '.dll'
     filter_from_dll:
-        - OriginalFilename|endswith: 
+        - OriginalFilename|endswith:
             - '.dll'
             - '.tmp'  # VSCode FP
-        - OriginalFilename|contains: 
+        - OriginalFilename:
```
Comment (frack113) [inline rules/windows/file_rename/file_rename_win_not_dll_to_dll.yml:21]: This is a `file_rename` from ETW kernel-file-rename not a `process_creation`.
You can see `OriginalFilename` not `OriginalFileName`
`N` make the diff 😄

File: rules/windows/process_creation/proc_creation_win_creation_mavinject_dll.yml [hunk 4]
```diff
@@ -18,11 +19,11 @@ logsource:
     product: windows
 detection:
     selection:
-        CommandLine|contains|all: 
+        CommandLine|contains|all:
             - ' /INJECTRUNNING'
             - '.dll' # space some time in the end
-        OriginalFileName|contains: mavinject
-    condition: selection 
+        OriginalFileName: mavinject
```
Comment (frack113) [inline rules/windows/process_creation/proc_creation_win_creation_mavinject_dll.yml:25]:
```
OriginalFileName: 
  - mavinject32.exe
  - mavinject64.exe
```

File: rules/windows/process_creation/proc_creation_win_susp_trolleyexpress_procdump.yml [hunk 5]
```diff
@@ -25,7 +26,7 @@ detection:
   renamed:
     Image|endswith: '\TrolleyExpress.exe'
   filter_renamed:
-    OriginalFileName|contains: 'CtxInstall'
+    OriginalFileName: 'CtxInstall'
```
Comment (frack113) [inline rules/windows/process_creation/proc_creation_win_susp_trolleyexpress_procdump.yml:29]: Don't have the binary so not sur it is `CtxInstall.exe` so keep the `|contains`

File: rules/windows/process_creation/proc_creation_win_wmi_spwns_powershell.yml [hunk 6]
```diff
@@ -7,19 +7,21 @@ references:
...
-        Image|endswith: '\powershell.exe'
+    selection_img:
+        - Image|endswith: '\powershell.exe'
+        - OriginalFileName: 'PowerShell.EXE'
     filter_null1:
         CommandLine: 'null'
     filter_null2:  # some backends need the null value in a separate expression
         CommandLine: null
-    condition: selection and not filter_null1 and not filter_null2
+    condition: all of selection* and not filter_null1 and not filter_null2
```
Comment (frack113) [inline rules/windows/process_creation/proc_creation_win_wmi_spwns_powershell.yml:24]: `condition: all of selection* and not all of filter_*`

**Review by frack113 — CHANGES_REQUESTED** [review]
Some remarks

**Review by frack113 — APPROVED** [review]

**Pull request merged**